File: 1980. Find Unique Binary String/main.py
from typing import List


class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        ARRAY_SIZE = 16
        MAX_COUNT = 1 << ARRAY_SIZE
        exists = [False] * MAX_COUNT

        for num in nums:
            index = int(num, 2)
            exists[index] = True

        val = 0
        while val < MAX_COUNT:
            if not exists[val]:
                break
            val += 1

        return bin(val)[2:].zfill(len(nums))


class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        ARRAY_SIZE = 16
        MAX_COUNT = 1 << ARRAY_SIZE
        exists = [False] * MAX_COUNT

        for num in nums:
            index = int(num, 2)
            exists[index] = True

        val = 0
        while val < MAX_COUNT:
            if not exists[val]:
                break
            val += 1

        return f"{val:0{len(nums)}b}"


class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        ARRAY_SIZE = 16
        MAX_COUNT = 1 << ARRAY_SIZE
        exists = [False] * MAX_COUNT

        for num in nums:
            index = int(num, 2)
            exists[index] = True

        val = 0
        while val < MAX_COUNT:
            if not exists[val]:
                break
            val += 1

        print(val)
        res = ""
        size = len(nums)
        for i in range(size - 1, -1, -1):
            res += chr(((val >> i) & 1) + ord("0"))

        return res


class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        res = ""

        for i in range(len(nums)):
            if nums[i][i] == "0":
                res += "1"
            else:
                res += "0"

        return res


class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        res = list()

        for i in range(len(nums)):
            if nums[i][i] == "0":
                res.append("1")
            else:
                res.append("0")

        return "".join(res)

File: 1980. Find Unique Binary String/main.cpp
#include <bitset>
#include <iostream>
#include <string>
#include <vector>

using std::bitset;
using std::string;
using std::vector;

#define ARRAY_SIZE 16

class Solution {
public:
    string findDifferentBinaryString(vector<string> &nums) {
        bool exists[1 << ARRAY_SIZE] = {false};
        size_t size = nums.size();
        for (size_t i = 0; i < size; i++) {
            bitset<32> bits(nums[i]);
            unsigned long value = bits.to_ulong();
            if (value >= 1 << size) {
                return "";
            }

            exists[value] = true;
        }

        size_t val = 0;
        while (val < 1 << size) {
            if (!exists[val]) {
                break;
            }
            val++;
        }

        std::bitset<32> bits(val);
        std::string binaryString = bits.to_string().substr(32 - size);

        return binaryString;
    }
};

#define ARRAY_SIZE 16

class Solution {
public:
    string findDifferentBinaryString(vector<string> &nums) {
        bool exists[1 << ARRAY_SIZE] = {false};
        size_t size = nums.size();
        for (size_t i = 0; i < size; i++) {
            unsigned int value = 0;
            for (const auto &c : nums[i]) {
                value = (value << 1) | (c - '0');
            }

            exists[value] = true;
        }

        size_t val = 0;
        while (val < 1 << size) {
            if (!exists[val]) {
                break;
            }
            val++;
        }

        string res;
        size_t index = 0;
        while (index < size) {
            char c = ((val >> (size - 1 - index)) & 1) + '0';
            res.push_back(c);
            index++;
        }

        return res;
    }
};

class Solution {
public:
    string findDifferentBinaryString(vector<string> &nums) {
        string res;
        size_t size = nums.size();

        for (size_t i = 0; i < size; i++) {
            if (nums[i][i] == '0') {
                res.push_back('1');
            } else {
                res.push_back('0');
            }
        }

        return res;
    }
};